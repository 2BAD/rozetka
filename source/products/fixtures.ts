export const PRODUCT_ID = 6802844
export const PRODUCT_IDS_ARRAY = [6802844, 9906369]
export const PRODUCT_ID_WITH_TAGS = 318463618
export const WRONG_PRODUCT_ID = 999999999
