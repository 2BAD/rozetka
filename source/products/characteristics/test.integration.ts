import { PRODUCT_ID, WRONG_PRODUCT_ID } from '../fixtures'
import { get } from './index'

describe('/products/characteristics', () => {
  describe('GET', () => {
    it('When asked for an existing product \n\t Then should be able to retrieve and parse its characteristics', async () => {
      expect.assertions(1)
      const r = await get(PRODUCT_ID)
      expect(r.data).toMatchSnapshot()
    })
    it('When asked for an non-existing product \n\t Then should return an Error object', async () => {
      expect.assertions(1)
      const r = await get(WRONG_PRODUCT_ID)
      expect(r.error).toMatchSnapshot()
    })
  })
})
