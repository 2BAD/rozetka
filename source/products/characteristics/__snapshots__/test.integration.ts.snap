// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/products/characteristics GET When asked for an existing product 
	 Then should be able to retrieve and parse its characteristics 1`] = `
Array [
  Object {
    "groupTitle": "",
    "group_id": 0,
    "group_order": 99999999,
    "options": Array [
      Object {
        "category_id": 4626923,
        "comparable": "main",
        "id": 147152,
        "in_short": 0,
        "name": "upakovka147152",
        "order": 3,
        "status": "active",
        "title": "Упаковка",
        "type": "List",
        "values": Array [
          Object {
            "href": "/voda/c4639872/upakovka147152=plastikovaya/",
            "title": "Пластиковая бутылка",
          },
        ],
      },
      Object {
        "category_id": 4625018,
        "comparable": "main",
        "id": 168958,
        "in_short": 0,
        "name": "stepen-gazirovaniya",
        "order": 21,
        "status": "active",
        "title": "Степень газирования",
        "type": "ComboBox",
        "values": Array [
          Object {
            "href": "/voda/c4639872/stepen-gazirovaniya=bez-gaza/",
            "title": "Без газа",
          },
        ],
      },
      Object {
        "category_id": 4633001,
        "comparable": "main",
        "id": 71581,
        "in_short": 0,
        "name": "objom-71581",
        "order": 24,
        "status": "active",
        "title": "Объем",
        "type": "ComboBox",
        "values": Array [
          Object {
            "href": "/voda/c4639872/objom-71581=0-5-0-99-l/",
            "title": "0.5 л",
          },
        ],
      },
      Object {
        "category_id": 4639872,
        "comparable": "main",
        "id": 168950,
        "in_short": 0,
        "name": "tip168950",
        "order": 27,
        "status": "active",
        "title": "Тип",
        "type": "List",
        "values": Array [
          Object {
            "href": "/voda/c4639872/tip168950=mineralnaya/",
            "title": "Минеральная",
          },
        ],
      },
      Object {
        "category_id": 0,
        "comparable": "main",
        "id": 92802,
        "in_short": 0,
        "name": "kolichestvo-predmetov-92802",
        "order": 60,
        "status": "active",
        "title": "Количество в упаковке, шт",
        "type": "ComboBox",
        "values": Array [
          Object {
            "href": "/voda/c4639872/kolichestvo-predmetov-92802=475541/",
            "title": "6",
          },
        ],
      },
      Object {
        "category_id": 4624997,
        "comparable": "disable",
        "id": 71434,
        "in_short": 0,
        "name": "srok-godnosti-71434",
        "order": 65,
        "status": "active",
        "title": "Срок хранения",
        "type": "TextInput",
        "values": Array [
          Object {
            "href": null,
            "title": "2 года",
          },
        ],
      },
      Object {
        "category_id": 0,
        "comparable": "main",
        "id": 98900,
        "in_short": 0,
        "name": "strana-proizvoditelj-tovara-90098",
        "order": 951,
        "status": "active",
        "title": "Страна производитель",
        "type": "ListValues",
        "values": Array [
          Object {
            "href": "/voda/c4639872/strana-proizvoditelj-tovara-90098=619938/",
            "title": "Франция",
          },
        ],
      },
    ],
  },
]
`;

exports[`/products/characteristics GET When asked for an non-existing product 
	 Then should return an Error object 1`] = `
[ZodError: [
  {
    "code": "custom",
    "message": "Bad response. The item you requested is not available.",
    "path": []
  }
]]
`;
